cvat-ai/cvat
# Publishing dev version of docker images

## Motivation and context

### What's new
- This PR adds the ability to build a development version of CVAT **without** building local images (which takes ~15 minutes)
- CVAT organization on DockerHub will contain another two tags for images `cvat/server:dev` and `cvat/ui:dev`
- PR also adds possibility to run specific version of CVAT e.g `v2.1.0`  
- To install development version of CVAT users can run this commands
```
export CVAT_VERSION="dev"
docker-compose -f docker-compose.yml up -d
```

### Limitations
- Users can only run the latest development version, which means there is no way to pull CVAT images from a particular development branch commit.
- Before pushing a development image to Docker Hub, we do not fully test them, we just run small sanity checks (main workflow), and only if these checks pass, we push the images.

### How has this been tested?
- Make sure the "classic" installation still works and make sure the pulled images are up to date with the latest CVAT.
- Make sure the "development" install works and make sure the pulled images are up to date the the development version of CVAT

### TO-DO
- [x] Add publishing docker images step for main workflow
- [x] Update docker-compose files

### Checklist

- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes -> tested this manually
~~- [ ] I have linked related issues ([read github docs](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)) -> I didn't find related issues~~
~~- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning)) -> there is no UI changes~~

### License

  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
